components/permission-provider/org.wso2.carbon.analytics.permissions/src/main/java/org/wso2/carbon/analytics/permissions/internal/dao/PermissionsDAO.java

```diff
@@ -81,6 +81,38 @@ public void addPermission(Permission permission) {
         }
     }
 
+    /**
+     * Check Permission already there.
+     *
+     * @param permission
+     */
+    public boolean isPermissionExists(Permission permission) {
+        boolean hasPermission = false;
+        Connection conn = null;
+        PreparedStatement ps = null;
+        ResultSet resultSet = null;
+        String query = "SELECT * FROM ROLE_PERMISSIONS WHERE APP_NAME = ? AND PERMISSION_STRING = ?";
+
+        try {
+            conn = getConnection();
+            ps = conn.prepareStatement(query);
+            ps.setString(1, permission.getAppName());
+            ps.setString(2, permission.getPermissionString());
+            if (log.isDebugEnabled()) {
+                log.debug("Executing query: " + query);
+            }
+            resultSet = ps.executeQuery();
+            while (resultSet.next()) {
+                hasPermission = true;
+            }
+        } catch (SQLException e) {
+            throw new PermissionException("Unable to check permissions exist. [Query=" + query + "]", e);
```

Comment:
We are just printing the query template we used as a basis of the PreparedStatement here (with the placeholder "?" values. Is this needed?

Reply:
Fixed 4911621